twine/settings.py

```diff
@@ -253,12 +253,20 @@ def _handle_authentication(self, username, password):
             username,
             self.repository_config
         )
-        self.password = utils.get_password(
-            self.repository_config['repository'],
-            self.username,
-            password,
-            self.repository_config,
-        )
+        if self.client_cert:
+            self.password = utils.get_password_no_prompt(
```

Comment:
Tracing this call, I see that this call will now look for passwords from the environment or config file or CLI, but not from keyring or a command-line prompt. Is a CLI-supplied or environment-supplied password ever relevant when a client cert is present? Should the `get_password` just be skipped entirely?

Reply:
Agree.